lib/api/2.0/nodes.js

```diff
@@ -61,24 +62,45 @@ var nodesGetPollersById = controller(function(req) {
 });
 
 var nodesGetWorkflowById = controller(function(req) {
-    return nodes.getNodeWorkflowById(req.swagger.params.identifier.value);
+    return workflowApiService.getWorkflowsByNodeId(req.swagger.params.identifier.value,
```

Comment:
I still think this should be changed to just get the required document from the query. If the worry is about being database agnostic, then I would suggest the query should be handled at the ORM layer rather than doing an iterative approach at this layer. Part of my concern is that I've seen developers in the past calling this route _a lot_ when building automation on top of us, primarily just to check if the node has an active workflow or not, and because doing it this way can potentially be so expensive, I think it's possible for us to run into a performance impact here.